### PR TITLE
feat!: Remove deprecated features, update upgrade guides

### PR DESCRIPTION
**Pull request by joeyparrish**
Below are the changelog entries for each deprecated feature removed by this commit.

-----

feat(config)!: `manifest.dash.defaultPresentationDelay` has been replaced by `manifest.defaultPresentationDelay` (deprecated in v3.0.0)

feat(config)!: Configuration of factories should be plain factory functions, not constructors; these will not be invoked with `new` (deprecated in v3.1.0)

feat(player)!: `shaka.Player.prototype.addTextTrack()` has been replaced by `addTextTrackAsync()`, which returns a `Promise` (deprecated in v3.1.0)

feat(ui)!: `shaka.ui.TrackLabelFormat` has been renamed to `shaka.ui.Overlay.TrackLabelFormat` (deprecated in v3.1.0)

feat(ui)!: `shaka.ui.FailReasonCode` has been renamed to `shaka.ui.Overlay.FailReasonCode` (deprecated in v3.1.0)

feat(offline)!: `shaka.offline.Storage.prototype.store()` returns `AbortableOperation` instead of `Promise` (deprecated in v3.0.0)

feat(offline)!: `shaka.offline.Storage.prototype.getStoreInProgress()` has been removed; concurrent operations are supported, so callers don't need to check this (deprecated in v3.0.0)

feat!: `shaka.util.Uint8ArrayUtils.equal` has been replaced by `shaka.util.BufferUtils.equal`, which can handle multiple types of buffers (deprecated in v3.0.0)

feat(manifest)!: `shaka.media.SegmentIndex.prototype.destroy()` has been replaced by `release()`, which is synchronous (deprecated in v3.0.0)

feat(manifest)!: `shaka.media.SegmentIterator.prototype.seek()`, which mutates the iterator, has been replaced by `shaka.media.SegmentIndex.getIteratorForTime()` (deprecated in v3.1.0)

feat(manifest)!: `shaka.media.SegmentIndex.prototype.merge()` has become private; use `mergeAndEvict()` instead (deprecated in v3.2.0)

feat(plugin)!: `AbrManager` plugins must implement the `playbackRateChanged()` method (deprecated in v3.0.0)

feat(plugin)!: `shaka.extern.Cue.prototype.spacer` has been replaced by the more clearly-named `lineBreak` (deprecated in v3.1.0)

feat(plugin)!: `IUIElement` plugins must have a `release()` method (not `destroy()`) (deprecated in v3.0.0)

### CODE REVIEW

**Comment by joeyparrish** [comment]
Ping!  @theodab, could you please look at this?  Merging this wouldn't require us to instantly launch v4, but it would be nice to have this done and ready.

File: docs/tutorials/upgrade.md [hunk 1]
```diff
@@ -1,17 +1,84 @@
...
+    - `shaka.ui.FailReasonCode` has been renamed to
+      `shaka.ui.Overlay.FailReasonCode` (deprecated in v3.1.0)
+
+  - Offline API changes:
+    - `shaka.offline.Storage.prototype.store()` returns `AbortableOperation`
+      instead of `Promise`; callers should change `.then()` to
+      `.promise.then()` and `await rv` to `await rv.promise` (deprecated in
+      v3.0.0)
+    - `shaka.offline.Storage.prototype.getStoreInProgress()` has been removed;
+      concurrent operations are supported, so callers don't need to check this
```
Comment (theodab) [inline docs/tutorials/upgrade.md:61]: Should probably specify concurrent operations are "now" supported, to make it clear that that was a change. As-is, it makes it sound like `getStoreInProgress` was an unnecessary feature we are removing, or something.

**Review by theodab — CHANGES_REQUESTED** [review]
LGTM otherwise.

File: docs/tutorials/upgrade.md (hunk 1 above)
Comment (joeyparrish) [reply docs/tutorials/upgrade.md]: Done, thanks!

**Review by theodab — APPROVED** [review]

**Pull request merged**